handlers.go

```diff
@@ -93,6 +90,22 @@ func loginSubmit(w http.ResponseWriter, r *http.Request) {
 		log.Println("Login failed -", err)
 		http.Redirect(w, r, "/login-fail", http.StatusSeeOther)
 	} else {
+		cookie, err := r.Cookie("c_user")
+		if err != nil {
+			fmt.Println("cookie was not found")
+			id, err := matcha.database.GetUserID("username", username)
+			if err != nil {
+				fmt.Println("getting user id failed -", err)
+			}
```

Comment:
Delete this, ass your new user feature is taking over the URL form sessions functionality completely.